src/graph/impl/KokkosGraph_Distance2Color_impl.hpp

```diff
@@ -40,899 +40,1615 @@
-
-    KOKKOS_INLINE_FUNCTION
-    void operator()(const nnz_lno_t i) const
+    /**
+     * Functor for VB algorithm speculative coloring without edge filtering.
+     * Single level parallelism
+     */
+    struct functorGreedyColorVB
     {
-      // Natural order
-      _vertexList(i) = i;
-    }
-  };  // struct functorInitList (end)
-
-
-
-  /**
-   * Functor for VB algorithm speculative coloring without edge filtering.
-   */
-  struct functorGreedyColor
-  {
-    nnz_lno_t                nv;                  // num vertices
-    const_lno_row_view_t     _idx;                // vertex degree list
-    const_lno_nnz_view_t     _adj;                // vertex adjacency list
-    const_clno_row_view_t    _t_idx;              // transpose vertex degree list
-    const_clno_nnz_view_t    _t_adj;              // transpose vertex adjacency list
-    color_view_type          _colors;             // vertex colors
-    nnz_lno_temp_work_view_t _vertexList;         //
-    nnz_lno_t                _vertexListLength;   //
-    nnz_lno_t                _chunkSize;          //
-
-    functorGreedyColor(nnz_lno_t                nv_,
-                       const_lno_row_view_t     xadj_,
-                       const_lno_nnz_view_t     adj_,
-                       const_clno_row_view_t    t_xadj_,
-                       const_clno_nnz_view_t    t_adj_,
-                       color_view_type          colors,
-                       nnz_lno_temp_work_view_t vertexList,
-                       nnz_lno_t                vertexListLength,
-                       nnz_lno_t                chunkSize)
-          : nv(nv_),
-            _idx(xadj_),
-            _adj(adj_),
-            _t_idx(t_xadj_),
-            _t_adj(t_adj_),
-            _colors(colors),
-            _vertexList(vertexList),
-            _vertexListLength(vertexListLength),
-            _chunkSize(chunkSize)
+        nnz_lno_type             nv;                     // num vertices
+        const_lno_row_view_type  _idx;                   // vertex degree list
+        const_lno_nnz_view_t     _adj;                   // vertex adjacency list
+        const_clno_row_view_t    _t_idx;                 // transpose vertex degree list
+        const_clno_nnz_view_t    _t_adj;                 // transpose vertex adjacency list
+        color_view_type          _colors;                // vertex colors
+        nnz_lno_temp_work_view_t _vertexList;            //
+        nnz_lno_type             _vertexListLength;      //
+        nnz_lno_type             _chunkSize;             //
+
+        functorGreedyColorVB(nnz_lno_type             nv_,
+                             const_lno_row_view_type  xadj_,
+                             const_lno_nnz_view_t     adj_,
+                             const_clno_row_view_t    t_xadj_,
+                             const_clno_nnz_view_t    t_adj_,
+                             color_view
```

Comment:
Lot of experimental stuff here ... Do we want to push ?

Reply:
@srajama1 This is part of the tight-bound-on-distance2-degree stuff. I've moved the `graph_compute_distance2_degree` function out of `KokkosGraph_Distance2Color.hpp` and into `KokkosGraph_Distance2Color_impl.hpp` so this thing only can get called if you're using `KokkosGraph::Impl` namespace is that ok or would it be better to try and move it into something in an `Experimental` namespace?

Reply:
If it is not exposed to user and we plan to clean it up soon, I am fine with it. Please file an issue so we don't forget.